src/language/class-modifiers.md

```diff
@@ -0,0 +1,240 @@
+its consumers.
+- Ensuring access to private members on instances of the final class will never
+throw a [`noSuchMethod`][] error.
+
+## `sealed`
+
+Use the `sealed` modifier to define a class that **cannot be constructed, extended,
+or implementated** outside of the library where the class is defined. 
+
+The purpose of the `sealed` modifier is to enable [_exhaustiveness checking_][]
+for switch cases.
+Any possible direct subtypes of a sealed type can only exist in the same library,
+so the compiler is always aware of the extent of a sealed type's hierarchy.
+
+This allows the compiler to alert you when a switch is not exhaustively handling
+all possible subtypes in its cases.
+
+```dart
+sealed class Vehicle {
+  String make;
+  String model;
+  void moveForward(int meters) { … }
+}
+
+class Car extends Vehicle { }
+class Truck implements Vehicle { }
+class Bicycle extends Vehicle { }
+
+// ...
+
+return switch (vehicle) {     // ERROR: The switch is missing a subtype of Vehicle
+  Car => 'vroom',
+  Truck => 'VROOOOMM'
+};
+```
+
+## `abstract`
+
+Use the `abstract` modifier to define a class that can’t be constructed.
+Abstract classes are useful for defining interfaces, often with some
+implementation. If you want your abstract class to appear to be instantiable,
+define a [factory constructor][].
+
+Abstract classes often have [abstract methods][].
+Here’s an example of declaring an abstract class that has an abstract
+method:
+
+<?code-excerpt "misc/lib/language_tour/classes/misc.dart (abstract)"?>
+```dart
+// This class is declared abstract and thus
+// can't be instantiated.
+abstract class AbstractContainer {
+  // Define constructors, fields, methods...
+
+  void updateChildren(); // Abstract method.
+}
+```
+
+## `class`, `mixin`, or `mixin class`?
+
+The `mixin` declaration defines a [mixin][]. The `mixin` _modifier_ can only be
+applied to a `class` declaration to create a _mixin class_: a class that can be
+treated both as a class and a mixed in. Regular classes cannot be mixed in.
+
+Mixins and mixin classes are meant to be mixed in, so they cannot be combined
+with the modifiers that prevent inheritance: `interface`, `final`, and `sealed`.
+
+## Combining modifiers
+
+It is possible to combine some modifiers for layered restrictions. For example, 
+an `abstract interface class` combines restrictions on instantiation and inheritance.
+This could be useful to define a "pure contract", or a class that doesn't update
+any preexisting state.
+
+Modifiers that are mutually exclusive cannot be combined. For example,
+`base` allows extending and disallows implementing, and `interface` allows
+implementing and disallows extending.
+
+Modifiers that are redundant cannot be combined. For example, mixins and sealed
+classes already cannot be constructed, so combining them with `abstract`, which
+disallows construction, is redundant. 
+
+The [syntax specification] documents the valid combinations and their capabilities.
```

Comment:
As with all writing, it depends on who the intended audience is.
I'd consider these documents as introductory. They are not exhaustive in defining or describing the language, so I would focus in the common use-cases, for the syntax, to *show* how it's used, not *specify* it.

(We have a secondary problem that we don't have a good, complete and readable exhaustive description of the language. The specification fails, at least, the "readable" part, because it mixes intricate semantic details with the "what does this language construct really do" text, and is very short on examples.)

Reply:
> We have a secondary problem that we don't have a good, complete and readable exhaustive description of the language

Yeah that issue is definitely on my radar. I'd like to start developing a "reference" type of documentation, especially now with patterns and class modifiers. Hopefully after this release 🤞

Reply:
I made an issue to track the need for further reference for this topic: https://github.com/dart-lang/site-www/issues/4783